lib/Consul/Check.pm

```diff
@@ -3,20 +3,56 @@ package Consul::Check;
 use namespace::autoclean;
 
 use Moo::Role;
-use Types::Standard qw(Str HashRef);
+use Types::Standard qw(Str HashRef Enum);
 
-has name  => ( is => 'ro', isa => Str, required => 1 );
-has id    => ( is => 'ro', isa => Str );
-has notes => ( is => 'ro', isa => Str );
+use Carp qw(croak);
+
+my $status_type = Enum['passing', 'warning', 'critical'];
+
+has name       => ( is => 'ro', isa => Str, required => 1 );
+has id         => ( is => 'ro', isa => Str );
+has notes      => ( is => 'ro', isa => Str );
+has service_id => ( is => 'ro', isa => Str );
+has status     => ( is => 'ro', isa => $status_type );
+has deregister_critical_service_after =>
+                  ( is => 'ro', isa => Str );
+
+sub smart_new {
```

Comment:
I would add though that I think `smart_new` in a role package (`Consul::Check`) is just weird, in location and naming.  Maybe a private method inside of `Consul::API::Agent` (assuming I at all talked you into keeping it).  I do like having it somewhere outside of the API package because it seems like a very general purpose coercion to me.